constraint-solver/src/quadratic_symbolic_expression.rs

```diff
@@ -434,10 +442,16 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display> QuadraticSymbolicExp
                     // tight good enough.
                     return Ok(ProcessResult::empty());
                 }
-                concrete_assignments.push(Effect::Assignment(
-                    variable.clone(),
-                    T::from(component >> exponent).into(),
-                ));
+                concrete_assignments.push(
+                    // TODO: This would be the right thing to do, but it
+                    // leads to failing tests...
+                    // assignment_if_satisfies_range_constraints(
+                    //     variable.clone(),
+                    //     T::from(component >> exponent).into(),
+                    //     range_constraints,
+                    // )?,
```

Comment:
I think it fails because of the `?` operator. We check the range constraints here all the time and I think we should get an `Error::ConstraintUnsatisfiable` in line 462 / 476 if the value does not fit the variable's range constraint.